libcni/api.go

```diff
@@ -165,33 +169,41 @@ func (c *CNIConfig) ensureExec() invoke.Exec {
 	return c.exec
 }
 
-func getResultCacheFilePath(netName string, rt *RuntimeConf) string {
+type cacheFileType string
+
+const (
+	cacheFileTypeResult cacheFileType = "results"
+	cacheFileTypeConfig cacheFileType = "config"
+)
+
+func getCacheFilePath(fileType cacheFileType, netName string, rt *RuntimeConf) string {
 	cacheDir := rt.CacheDir
 	if cacheDir == "" {
 		cacheDir = CacheDir
 	}
-	return filepath.Join(cacheDir, "results", fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName))
+	fName := fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName)
+	return filepath.Join(cacheDir, string(fileType), base64.StdEncoding.EncodeToString([]byte(fName)))
```

Comment:
What if **rt.CacheDir** changes ? We will not get the **previous** cached file in a **new** path. Maybe we can have a fixed cache dir here.

Reply:
I would say that runtimes simply shouldn't do that, but it's an interesting point. I think we put CacheDir into RuntimeConfig because then it's not a required argument for NewCNIConfig() and thus CNI users don't have to change anything unless they want a custom cache dir.

Reply:
Yes, I also agree on that runtime should not change cache dir freely.
So maybe there is another way, we can put cache dir from runtimeConfig to CNIConfig, and add a new constructor `func NewCNIConfig(path []string, exec invoke.Exec, cacheDir string) *CNIConfig`, if we use the old constructor `NewCNIConfig(path []string, exec invoke.Exec) *CNIConfig`, the cache dir will be a default value. Once cache dir is set in CNIConfig, it won't be changed by any CNI user.